core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -67,8 +67,11 @@ private class AsyncEventQueue(
   /** A counter for dropped events. It will be reset every time we log it. */
   private val droppedEventsCounter = new AtomicLong(0L)
 
+  /** A counter to keep number of dropped events last time it was logged */
+  private var lastDroppedEventsCounter: Long = 0L
+
   /** When `droppedEventsCounter` was logged last time in milliseconds. */
-  @volatile private var lastReportTimestamp = 0L
+  @volatile private val lastReportTimestamp = new AtomicLong(0L)
```

Comment:
No longer need to have `@volatile` as it becomes val.

Reply:
Yes, I'ill update it.